code/datums/components/_component.dm

```diff
@@ -281,10 +285,13 @@
 	if(target.parent)
 		target.RemoveComponent()
 	target.parent = src
-	if(target.PostTransfer() == COMPONENT_INCOMPATIBLE)
-		var/c_type = target.type
-		qdel(target)
-		CRASH("Incompatible [c_type] transfer attempt to a [type]!")
+	var/result = target.PostTransfer()
+	switch(result)
+		if(COMPONENT_INCOMPATIBLE)
+			var/c_type = target.type
+			qdel(target)
+			stack_trace("Incompatible [c_type] transfer attempt to a [type]!")

```

Comment:
Shouldn't this still be a crash ? It shouldn't try to rejoin deleted component.

Reply:
oh god it's supposed to be deleting itself, not the target atom

Reply:
How's this going.

Reply:
I have done the needful

Reply:
I think you got confused, src here is the holder atom so it was actually correct before.

Reply:
aheiouadshfslkdajfl;fj